File: LeeCode/Go/3153. 所有数对中数位不同之和.go
import "strconv"
// 超出时间限制
func sumDigitDifferences(nums []int) int64 {
	var account int64
	for i := 0; i < len(nums); i++ {
		for j := i + 1; j < len(nums); j++ {
			account += count(nums[i], nums[j])
		}
	}
	return account
}

func count(value1 int, value2 int) int64 {
	newvalue1 := strconv.Itoa(value1)
	newvalue2 := strconv.Itoa(value2)
	lens := len(newvalue1)
	var count int64
	for i := 0; i < lens; i++ {
		if newvalue1[i] != newvalue2[i] {
			count++
		}
	}
	return count
}

File: LeeCode/JavaScript/3153. 所有数对中数位不同之和.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var sumDigitDifferences = function (nums) {
    let account = 0
    const len = nums.length
    for (let i = 0; i < len; i++) {
        for (let j = i + 1; j < len; j++) {
            account += count(nums[i], nums[j])
        }
    }
    return account
};

const count = (value1, value2) => {
    if (value1 == value2) return 0
    const value2Str = String(value2)
    const value1Str = String(value1)
    const lens = value1Str.length
    let count = 0
    for (let i = 0; i < lens; i++) {
        if (value1Str[i] != value2Str[i]) {
            count++
        }
    }
    return count
}